pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -379,9 +389,16 @@ async def _prepare_request_parameters(
     """Build tools and create an agent model."""
     output_schema = ctx.deps.output_schema
 
-    prompted_output_template = (
-        output_schema.template if isinstance(output_schema, _output.PromptedOutputSchema) else None
-    )
+    # Get the prompted output template with precedence:
+    # PromptConfig template > PromptedOutput.template > model profile default (handled downstream)
+    prompted_output_template: str | None = None
+    if isinstance(output_schema, _output.PromptedOutputSchema):
+        prompt_config = ctx.deps.prompt_config
+        templates = prompt_config.templates if prompt_config else None
+        config_template = templates.prompted_output_template if templates else None
+        # Only use config_template if explicitly set, otherwise preserve original behavior
+        # and let the model apply its profile-specific default
+        prompted_output_template = config_template or output_schema.template
```

Comment:
I think this is easier to follow than the multiple `x if y.foo else None` constructions:

```suggestion
    if isinstance(output_schema, _output.PromptedOutputSchema):
        if (
            (prompt_config := ctx.deps.prompt_config)
            and (templates := prompt_config.templates)
            and (template := templates.prompted_output_template)
        ):
            prompted_output_template = template
        else:
            prompted_output_template = output_schema.template
```